hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/placement/MultiNodePolicySpec.java

```diff
@@ -22,14 +22,25 @@
  */
 public class MultiNodePolicySpec {
 
+  private String policyName;
   private String policyClassName;
   private long sortingInterval;
 
-  public MultiNodePolicySpec(String policyClassName, long timeout) {
+  public MultiNodePolicySpec(String policyName, String policyClassName,
```

Comment:
why both policyName and policyClassName is required here ?

Reply:
In MultiNodeSorter#initPolicy, Policy instance will be created based on policyClassName, and policyName will be used to get conf belong to this policy instance in MultiComparatorPolicy#setConf. This is the only way for every policy instance to know which configuration belong to it, another way is to update the policy interface that I prefer not to use. If there are better approaches, feel free to propose them.